cli/tests/test_check_env.py

```diff
@@ -28,7 +30,7 @@ def test_check_env():
     - current test env file matches the sample env file
     """
     runner = CliRunner()
-    result = runner.invoke(check_env)
+    result = runner.invoke(check_env, args=["--error"])
```

Comment:
this test is [failing](https://github.com/UCLH-Foundry/PIXL/actions/runs/9956945495/job/27507918188?pr=419#step:5:131) because none of the environment variables in the sample env file are set. How best to set them for the test? Or should we create a test sample env file in the same way `test_check_env_fails` does?

Reply:
ah this is #410, I've updated the description to say that this fixes it

Reply:
> Can we point to an env file where this has been set, like the system tests one?

oh good idea! I've done that now, although not all variables are set in that file so the rest are now set in the `conftest.py`

> Probably should be one failing test and one passing test rather than two in the same test?

Sorry I don't really follow what you mean?

Reply:
this tests feels like a positive and a negative test smooshed together. Feels like what we should do is to have one postitive test
```
result = runner.invoke(check_env, args=["--error"])
```

then another with the negative test 
```
tmp_sample_env_file.write_text("NONEXISTENT_VARIABLE=")

    runner = CliRunner()
    with pytest.raises(UndefinedValueError, match="not found"):
        runner.invoke(
            check_env,
            args=["--error", "--sample-env-file", tmp_sample_env_file],
            catch_exceptions=False,
        )
```

Reply:
Ah, I think that's how it currently is (`test_check_env` for the positive, `test_check_env_fails` for the negative):
https://github.com/UCLH-Foundry/PIXL/blob/285d0348444627d4fa7ceaf6232c946fe9ceabfe/cli/tests/test_check_env.py#L26-L50
Is it just the diff collapsing lines and making it look like one test? Sorry if I'm still not following here...

Reply:
ha! I could have sworn I checked for that but clearly not. Ignore me

Reply:
ha, no worries 😄 thanks for the review!